### PR TITLE
[OpenVINO] Unify torch representation for CausalConv1D

### PR DESCRIPTION
**Pull request by rkazants**
# What does this PR do?

Unify torch representation for CausalConv1 that will be used among all hybrid models. This is needed to get unified sub-graph that will be fused into the single operation.

## Before submitting
- [N/A] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [N/A] Did you make sure to update the documentation with your changes?
- [ it already exists] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/exporters/openvino/model_patcher.py [hunk 1]
```diff
@@ -7085,35 +7103,16 @@ def lfm2_short_conv_forward_patched(
...
-        is_decoding = torch.tensor(seqlen == 1).to(x.dtype)
-        conv_out = conv_out_dec * is_decoding + conv_out_prefill * (1.0 - is_decoding)
-        new_conv_state = conv_state_dec * is_decoding + conv_state_prefill * (1.0 - is_decoding)
-        past_key_values.conv_cache[layer_idx].copy_(new_conv_state)
-    else:
-        conv_out = self.conv(Bx)[..., :seqlen]
+    layer_idx = past_key_values.conv_layer_idx_mapping[self.layer_idx]
+    conv_state = past_key_values.conv_cache[layer_idx]
+    conv_out, new_conv_state = ov_causal_conv1d(conv_state, Bx, self.conv.weight, self.conv.bias)
+    past_key_values.conv_cache[layer_idx].copy_(new_conv_state)
```
[the remark this answers is not in the document]
Comment (rkazants) [reply optimum/exporters/openvino/model_patcher.py]: Fixed.

File: optimum/exporters/openvino/model_patcher.py [hunk 2]
```diff
@@ -7085,35 +7103,16 @@ def lfm2_short_conv_forward_patched(
...
-        conv_out_prefill = self.conv(Bx)[..., :seqlen]
-
-        is_decoding = torch.tensor(seqlen == 1).to(x.dtype)
-        conv_out = conv_out_dec * is_decoding + conv_out_prefill * (1.0 - is_decoding)
-        new_conv_state = conv_state_dec * is_decoding + conv_state_prefill * (1.0 - is_decoding)
-        past_key_values.conv_cache[layer_idx].copy_(new_conv_state)
-    else:
-        conv_out = self.conv(Bx)[..., :seqlen]
+    layer_idx = past_key_values.conv_layer_idx_mapping[self.layer_idx]
+    conv_state = past_key_values.conv_cache[layer_idx]
```
[the remark this answers is not in the document]
Comment (rkazants) [reply optimum/exporters/openvino/model_patcher.py]: `cache_position` input is not needed to `CausalConv1d` operation. It is something custom introduced into `lfm2` model. Let me explain why is it so?

There are two cases:
1. a number of tokens in prefill greater or equal to `conv_kernel_size` (`prefill_seq_len >= conv_kernel_size`). All cache positions will complete all positions in convolution window (0, 1, ..., seq_len-1), then in decoding stage position `pos >= seq_len` will be clamped into `conv_kernel_size-1` so that it will be placed to the tail of `conv_state` and the very first token will be removed from `conv_state`
2. a number of tokens in prefill less than `conv_kernel_size` (`prefill_seq_len < conv_kernel_size`). We assume that `conv_kernel_size` is small (I see 3 (lfm2, lfm2-moe), 4 (zamba2, granite-4.0-hybrid, qwen3-next)). Hovewer, having 2, 3 tokens in prefill stage is kind rare case. Also, I did accuracy testing with my changes and it shows exact the same accuracy value. Besides, I experimented with input prompts having one, two tokens and generated output is identical to what we have without these changes.

**Comment by rkazants** [comment]
> please check the copilot comments as they seem to raise some valid points

Hi Ilyas, I resolved these comments. Please check.
